samples/performance/hpp_pipeline_cache/hpp_pipeline_cache.cpp

```diff
@@ -0,0 +1,167 @@
+/* Copyright (c) 2022, NVIDIA CORPORATION. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "hpp_pipeline_cache.h"
+
+#include "common/hpp_utils.h"
+#include "hpp_gui.h"
+#include "rendering/subpasses/hpp_forward_subpass.h"
+
+HPPPipelineCache::HPPPipelineCache()
+{
+	auto &config = get_configuration();
+
+	config.insert<vkb::BoolSetting>(0, enable_pipeline_cache, true);
+	config.insert<vkb::BoolSetting>(1, enable_pipeline_cache, false);
+}
+
+HPPPipelineCache::~HPPPipelineCache()
+{
+	if (pipeline_cache)
+	{
+		/* Get data of pipeline cache */
+		std::vector<uint8_t> data = device->get_handle().getPipelineCacheData(pipeline_cache);
+
+		/* Write pipeline cache data to a file in binary format */
+		vkb::fs::write_temp(data, "pipeline_cache.data");
+
+		/* Destroy Vulkan pipeline cache */
+		device->get_handle().destroyPipelineCache(pipeline_cache);
+	}
+
+	vkb::fs::write_temp(device->get_resource_cache().serialize(), "cache.data");
+}
+
+bool HPPPipelineCache::prepare(vkb::platform::HPPPlatform &platform)
+{
+	if (!HPPVulkanSample::prepare(platform))
+	{
+		return false;
+	}
+
+	/* Try to read pipeline cache file if exists */
+	std::vector<uint8_t> pipeline_data;
+	try
+	{
+		pipeline_data = vkb::fs::read_temp("pipeline_cache.data");
+	}
+	catch (std::runtime_error &ex)
+	{
+		LOGW("No pipeline cache found. {}", ex.what());
+	}
+
+	/* Add initial pipeline cache data from the cached file */
+	// Note: unfortunately, no compiler can detect the correct data type from the "natural" constructor here:
+	//	vk::PipelineCacheCreateInfo pipeline_cache_create_info({}, pipeline_data);
+	// This is due to the templatized second argument. You could help the compiler by a cast
```

Comment:
There's a mix of C and C++ style comments all over this file. Might be nice if we were consistent.

Reply:
Made all comments here C-style comments